pkg/services/applicationv2/fetcher.go

```diff
@@ -0,0 +1,107 @@
+package applicationv2
+
+import (
+	"context"
+	"errors"
+
+	wego "github.com/weaveworks/weave-gitops/api/v1alpha1"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+	"github.com/weaveworks/weave-gitops/pkg/models"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+var ErrNotFound = errors.New("not found")
+
+//counterfeiter:generate . Fetcher
+type Fetcher interface {
+	Get(ctx context.Context, name string, namespace string) (models.Application, error)
+	List(ctx context.Context, namespace string) ([]models.Application, error)
+}
+
+func NewFetcher(k8s client.Client) Fetcher {
+	return fetcher{
+		k8s: k8s,
+	}
+}
+
+type fetcher struct {
+	k8s client.Client
+}
+
+func (f fetcher) Get(ctx context.Context, name string, namespace string) (models.Application, error) {
+	app := &wego.Application{}
+
+	err := f.k8s.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, app)
+
+	if apierrors.IsNotFound(err) {
+		return models.Application{}, ErrNotFound
+	} else if err != nil {
+		return models.Application{}, err
+	}
+
+	a, err := translateApp(*app)
+	if err != nil {
+		return models.Application{}, err
+	}
+
+	return a, nil
+}
+
+func (f fetcher) List(ctx context.Context, namespace string) ([]models.Application, error) {
```

Comment:
Minor: namespace isn't used in this function remove it as a parameter.

Reply:
I think it needs to be used :thinking:

Reply:
Yeah I have added it in and added a test to cover it.